### PR TITLE
Cleaned up moniker versioning for ASP.NET Core SignalR configuration

### LINKED ISSUE #15666: Reorg MD by monikers
See #15657

Version in our suggested approach. Add a 3.1 section and add warning, this topic is not updated to 3.1

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: de8cdc99-48bb-3266-05bb-eda98f59f2e2
* Version Independent ID: 66d252c3-6300-7e28-9aa0-ef2ff66e7a76
* Content: [ASP.NET Core SignalR configuration](https://docs.microsoft.com/en-us/aspnet/core/signalr/configuration?view=aspnetcore-3.0&tabs=dotnet#feedback)
* Content Source: [aspnetcore/signalr/configuration.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/signalr/configuration.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-signalr**
* GitHub Login: @bradygaster
* Microsoft Alias: **bradyg**

### PR DESCRIPTION
**Pull request by wadepickett**
[Internal Review](https://review.docs.microsoft.com/en-us/aspnet/core/signalr/configuration?view=aspnetcore-2.1&branch=pr-en-us-16060)

Fixes #15666 

Reorganized moniker sections and condensed where needed for file:
signalr/configuration.md

### CODE REVIEW

File: aspnetcore/signalr/configuration.md [hunk 1]
```diff
@@ -15,6 +15,13 @@ uid: signalr/configuration
 
 ASP.NET Core SignalR supports two protocols for encoding messages: [JSON](https://www.json.org/) and [MessagePack](https://msgpack.org/index.html). Each protocol has serialization configuration options.
 
+::: moniker range="= aspnetcore-3.1"
```
Comment (Rick-Anderson) [inline aspnetcore/signalr/configuration.md]: Remove this warning section

**Comment by Rick-Anderson** [comment]
Why are there 10 `::: moniker range`? You should only have 2.

**Comment by Rick-Anderson** [comment]
```
::: moniker range=">= aspnetcore-3.0"
::: moniker-end

::: moniker range="< aspnetcore-3.0"
::: moniker-end
```

**Comment by wadepickett** [comment]
@Rick-Anderson, OK sorry.  I thought we wanted to keep the original intent of the monikers in there, but organized they way we are supposed to do them.  I didn't realize we were getting rid of 1.0.  I'll get rid of it.

**Comment by Rick-Anderson** [comment]
Even if you kept 1.0, you'd only need 3 of `::: moniker range=`

**Comment by wadepickett** [comment]
@Rick-Anderson ,

I had to add = 3.1 in there if that is what you are referring to, so it would just show up in 3.1.  It seemed odd to call it out on the 3.0 doc.  I'll remove 3.1.

**Comment by wadepickett** [comment]
@Rick-Anderson,  I see I misunderstood what you were saying.   I see there are additional <3.0 monikers in the bottom half that I should have removed..   There are the = 2.2 and =2.1 that I need to leave in the bottom half (<3.0) for listed config options that are specific to those versions. Since I can't  nest moniker ranges ( At least I was told I can't, maybe that is wrong?) I have to pick back up with another <3.0 afterwards.    I will remove the 3.1 and  the two gratuitous <3.0 ranges left in from the original and   then maybe we can g over the update together.  I  may not be understanding what I am allowed to do and not do for nested ranges.  Thanks.

::: moniker range="< aspnetcore-3.0"
::: moniker-end

**Comment by wadepickett** [comment]
@Rick-Anderson 

Below is what I have in the PR right now.  I can't see a way to get by with less that this unless we ditch some version specific information or I am allowed to nest ranges somehow which is as far as I know we can't.  Thanks for the help and sorry to drag you through a teaching moment.

>= 3.0
end

< 3.0
end

= 2.2   ( specific options table for 2.2 only  )
end

= 2.1   ( specific options table for 2.1 only )
end

< 3.0   (Back to including all versions up to 2.2. after we had to do the 2.2 vs 2.1 specific info.)
end

< 2.2  ( specific options for 2.1 or lower )
end

<3.0 Back to including all versions up to 2.2.  after we had to do <2.2 specific info.
end

**Comment by Rick-Anderson** [comment]
No, I want

```
::: moniker range=">= aspnetcore-3.0"
// This will be 3.1 in another PR
::: moniker-end

::: moniker range="= aspnetcore-2.2"
::: moniker-end

::: moniker range="<aspnetcore-2.2"
::: moniker-end
```

My guess is they are missing specific options for 3.0.  I would add in the 2.2 table to the 3.0 section and call that out for review.

**Comment by wadepickett** [comment]
@Rick-Anderson,  to rephrase what you want to make sure I am on the same page.

You want three copies of the of the doc.  One that is 3.0 specific, one that is 2.2 specific and one that is <2.2 specific.

Correct?  If so no problem, I'll change.

Where the confusion was coming from on my part:
When you asked much earlier for this:
     ::: moniker range=">= aspnetcore-3.0"
     ::: moniker-end

     ::: moniker range="< aspnetcore-3.0"
     ::: moniker-end

... I thought you essentially wanted just two copies of the entire doc, which forced me to drop in sub version clauses for areas of the 2nd copy.  :)

**Comment by Rick-Anderson** [comment]
One copy per version is our policy - 3 copies, 3 versions.

**Comment by wadepickett** [comment]
Has problems, not ready.   Issues with a dupe table tabs then ended up in there when trying to sort out the versioning.  Also moving H1 above the first moniker....

File: aspnetcore/signalr/configuration.md [hunk 2]
```diff
@@ -5,18 +5,19 @@ description: Learn how to configure ASP.NET Core SignalR apps.
 monikerRange: '>= aspnetcore-2.1'
 ms.author: bradyg
 ms.custom: mvc
-ms.date: 11/12/2019
+ms.date: 12/09/2019
```
Comment (Rick-Anderson) [inline aspnetcore/signalr/configuration.md]: ms.date: 12/11/2019

**Comment by wadepickett** [comment]
Cleaned it up.  Ready.

**Comment by Rick-Anderson** [comment]
Compare each version side by side with published doc. Quick check found the following in 2.1 version

image

Let me know when it passes comparison and I'll take another look.

**Comment by Rick-Anderson** [comment]
3.0 start to differ here
image

**Comment by wadepickett** [comment]
@Rick-Anderson, thanks. Yes,I had done a  comparison, several.  This was fairly messy to sort out.  I spotted and made a note about the item you first mentioned early on about websockets. That to me seemed to be in the wrong place in the original:
"In this version of the Java client websockets is the only available transport."  

I was going to call in the PM to look it over as you had suggested once you looked through it.  I'm checking back through it line for line again right now for every version against the original.  Thanks for spotting it was off.  I did actually check against the original but still made a wrong assumption.  Hopefully I didn't miss more which would be shameful.

**Comment by wadepickett** [comment]
@Rick-Anderson,
I don't get the same diff you show for 3.0 versions.  They checkout fine with no diff.  
image

It looks like you might have had one page set to 3.0 and one page set to 2.2 accidentally.  I do recreate your screenshot if I do that.

I  did correct the "In this version of the Java client websockets " line that was showing up in 2.1 review but was not in the original published 2.1.  Thanks.  Next patch has that.

**Comment by wadepickett** [comment]
@RickAndMSFT , ready for review.  Thanks.  See my comments and screenshot on 3.0 comparison.  I did a visual check on all versions side by side x each tab.  Plus did a diff tool check against rendered text for all versions.

**Comment by Rick-Anderson** [comment]
Are the 3.0 versions identical? I thought the table was missing from the 3.0 version.

**Comment by Rick-Anderson** [comment]
@wadepickett  can we S&M this or do you have questions for the PU?

**Comment by wadepickett** [comment]
Squash and merge.  No questions for PU.

The options table for 3.0 was clear after looking back through it several times.  

I could not find a table difference between the original 3.0 and my PR 3.0.  There is only some corrected spelling for that section.  Thanks again for your help and guidance.

**Comment by Rick-Anderson** [comment]
@bradygaster @anurse @BrennanConroy

I had @wadepickett  rearrange this topic as follows:

```
::: moniker range=">= aspnetcore-3.0"
::: moniker-end

::: moniker range="= aspnetcore-2.2"
::: moniker-end

::: moniker range="<= aspnetcore-2.2"
::: moniker-end
```

Otherwise it's too hard to reason about how to update the topic.

When updating to 3.1, I wouldn't keep the 3.0 version. We have zero demand for 3.0 versions. Just update to 3.1.

2.1 should be kept. If there's a 2.2 version, might as well keep it too.

**Pull request merged**